From 0f813df9eb0a2b96b946b13993b6423ba560da7f Mon Sep 17 00:00:00 2001
From: Yong Zhao <yong.zhao@amd.com>
Date: Mon, 31 Oct 2016 18:55:44 -0400
Subject: [PATCH 1907/2094] drm/amdkfd: Remove useless
 amdgpu_vm_clear_invalids() calls

amdgpu_vm_clear_invalids() has no effect to the KFD use cases.

Change-Id: I6d68d89c999086645ce7c543a7ee74a4928703b9
Signed-off-by: Yong Zhao <yong.zhao@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 2b56ea3..ff3be96 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -855,20 +855,11 @@ static int map_bo_to_gpuvm(struct amdgpu_device *adev, struct amdgpu_bo *bo,
 
 	amdgpu_sync_fence(adev, sync, bo_va->last_pt_update);
 
-	ret = amdgpu_vm_clear_invalids(adev, vm, sync);
-	if (ret != 0) {
-		pr_err("amdkfd: Failed to radeon_vm_clear_invalids\n");
-		goto err_failed_to_vm_clear_invalids;
-	}
-
 	/* Remove PTs from LRU list (reservation removed PD only) */
 	amdgpu_vm_move_pt_bos_in_lru(adev, vm);
 
 	return 0;
 
-err_failed_to_vm_clear_invalids:
-	amdgpu_vm_bo_update(adev, bo_va, NULL);
-	amdgpu_sync_fence(adev, sync, bo_va->last_pt_update);
 err_unpin_bo:
 	amdgpu_amdkfd_bo_invalidate(bo);
 	return ret;
@@ -1332,8 +1323,6 @@ static int unmap_bo_from_gpuvm(struct amdgpu_device *adev,
 	amdgpu_vm_bo_update(adev, bo_va, NULL);
 	amdgpu_sync_fence(adev, sync, bo_va->last_pt_update);
 
-	amdgpu_vm_clear_invalids(adev, vm, sync);
-
 	amdgpu_amdkfd_bo_invalidate(bo);
 
 	return 0;
-- 
2.7.4

